pkg/tests/apis/folder/folders_test.go

```diff
@@ -926,3 +926,134 @@ func testDescription(description string, expectedErr error) string {
 		return description
 	}
 }
+
+// TestFoldersGetAPIEndpointK8S is the counterpart of pkg/api/folder_test.go TestFolderGetAPIEndpoint
+func TestFoldersGetAPIEndpointK8S(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping integration test")
+	}
+
+	type testCase struct {
+		description    string
+		expectedCode   int
+		params         string
+		createFolders  []string
+		expectedOutput []dtos.FolderSearchHit
+		permissions    []resourcepermissions.SetResourcePermissionCommand
+	}
+
+	folderCreatePermission := []resourcepermissions.SetResourcePermissionCommand{
+		{
+			Actions:           []string{"folders:create"},
+			Resource:          "folders",
+			ResourceAttribute: "uid",
+			ResourceID:        "*",
+		},
+	}
+
+	folder1 := "{ \"uid\": \"foo\", \"title\": \"Folder 1\"}"
+	folder2 := "{ \"uid\": \"bar\", \"title\": \"Folder 2\", \"parentUid\": \"foo\"}"
+	folder3 := "{ \"uid\": \"qux\", \"title\": \"Folder 3\"}"
+
+	tcs := []testCase{
+		{
+			description: "listing folders at root level succeeds",
+			createFolders: []string{
+				folder1,
+				folder2,
+				folder3,
+			},
+			expectedCode: http.StatusOK,
+			expectedOutput: []dtos.FolderSearchHit{
+				dtos.FolderSearchHit{UID: "foo", Title: "Folder 1"},
+				dtos.FolderSearchHit{UID: "qux", Title: "Folder 3"},
+			},
+			permissions: folderCreatePermission,
+		},
+		{
+			description: "listing subfolders succeeds",
+			createFolders: []string{
+				folder1,
+				folder2,
+				folder3,
+			},
+			params:       "?parentUid=foo",
+			expectedCode: http.StatusOK,
+			expectedOutput: []dtos.FolderSearchHit{
+				dtos.FolderSearchHit{UID: "bar", Title: "Folder 2", ParentUID: "foo"},
+			},
+			permissions: folderCreatePermission,
+		},
+	}
+
+	// test on all dualwriter modes
+	for mode := 1; mode <= 4; mode++ {
```

Comment:
Nice!

Reply:
🤔  why dont use this [pattern](https://github.com/grafana/grafana/blob/main/pkg/tests/apis/folder/folders_test.go#L127) when writing the tests? Doesn't it get confusing?

Reply:
I had actually created this [issue](https://github.com/grafana/grafana-org/issues/271) to cover all dual writer modes like [playlists](https://github.com/grafana/grafana/blob/main/pkg/tests/apis/playlist/playlist_test.go) is doing (and how we started to write folder tests in this file).

Reply:
> 🤔 why dont use this [pattern](https://github.com/grafana/grafana/blob/main/pkg/tests/apis/folder/folders_test.go#L127) when writing the tests? Doesn't it get confusing?

Hi @leonorfmartins,  the [pattern](https://github.com/grafana/grafana/blob/b0fe898fa1db3dd6524517741146c9c220b0023f/pkg/tests/apis/folder/folders_test.go#L127) you mentioned is testing the requests to the API Server directly. On the other hand `TestFoldersGetAPIEndpointK8S`, which is being introduced by this PR cares about testing the k8s version of the `GET /api/folders` endpoint. 

They are not testing the same thing.

On the other hand, your comment made me realize that we need to extend `TestIntegrationFoldersApp` to include CRUD tests for Listing folders.

Reply:
@leonorfmartins I added a test for List Folders to `TestIntegrationFoldersApp`